cm2350/e200z7.py

```diff
@@ -657,12 +652,13 @@ def stepi(self):
         """
         self.processIO()
 
+        pc = self.getProgramCounter()
+
```

Comment:
Why move it outside the try clause?  Before `checkException()`?

Reply:
the reason was to ensure that the `pc` variable was available to all of the exception handlers. I added logging the PC to the DebugException handler and when `getProgramCounter` was called after `checkException` then that exception handler caused an error. it seemed simple enough to move it ahead of the other functions to ensure that data is available for any exceptions or logging.

Reply:
Agreed on the logic.  Why outside the try/except clauses?

Reply:
I don't think I had a good reason for moving it outside the try/except clause. it didn't seem like it would matter if it was inside them or not.

Reply:
fair.